osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -253,7 +255,11 @@ public TModel Import(ArchiveReader archive)
 
                 model.Hash = computeHash(archive);
 
-                return Import(model, archive);
+                return await Import(model, archive, cancellationToken);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
             }
             catch (Exception e)
             {
```

Comment:
At the end of this method, there is:
`return null;`
I think it is a bad idea, the code should either:
1. rethrow the exception after logging
2. return a completed task, something like [this](https://stackoverflow.com/a/4246063/4215389) or this: `return Task.FromResult<TModel>(null).Result;`

Reply:
What's the issue with the way it's written? I'm not sure how solution 2 is any different.

Reply:
You are right, the second option is the same as returning `null`. At first I thought the method signature was `public Task<TModel> Import` without the `async` so the could would have returned a `null` Task which of course could lead to errors, but I was wrong. Excuse me.

Reply:
That said, the null return logic is very sloppy. We may want to change this in the future (for all public methods in this class).